turbopack/crates/turbo-tasks-backend/src/lmdb_backing_storage.rs

```diff
@@ -56,13 +60,34 @@ fn as_u32<E: Error + Send + Sync + 'static>(result: Result<&[u8], E>) -> Result<
     Ok(n)
 }
 
+struct ThreadLocalReadTransactionsContainer(UnsafeCell<SmallVec<[RoTransaction<'static>; 4]>>);
+
+impl ThreadLocalReadTransactionsContainer {
+    unsafe fn pop(&self) -> Option<RoTransaction<'static>> {
+        let vec = unsafe { &mut *self.0.get() };
+        vec.pop()
+    }
+
+    unsafe fn push(&self, tx: RoTransaction<'static>) {
+        let vec = unsafe { &mut *self.0.get() };
+        vec.push(tx)
+    }
+}
+
+// Safety: It's safe to send RoTransaction between threads, but the types don't allow that.
```

Comment:
A bit more detail for the comment:

```suggestion
// Safety: It's safe to send `RoTransaction` between threads as we construct `Environment` with
// `EnvironmentFlags::NO_TLS`, but the types don't allow that.
```